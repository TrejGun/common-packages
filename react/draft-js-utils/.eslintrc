{
  "extends": [
    "@gemunion/eslint-config/presets/tsx",
    "@gemunion/eslint-config/tests/jest"
  ],
  "parserOptions": {
    "project": "./tsconfig.test.json"
  },
  "root": true
}
