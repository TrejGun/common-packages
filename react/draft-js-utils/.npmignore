/@types
/src
/tsconfig.json
/jest.config.json
/setup-jest.ts
**/*.test.*
