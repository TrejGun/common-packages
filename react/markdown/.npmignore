/src
/@types
/tsconfig.json
/tsconfig.test.json
/*.stories.*
