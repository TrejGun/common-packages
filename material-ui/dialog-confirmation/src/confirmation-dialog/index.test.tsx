import React from "react";
import { IntlProvider } from "react-intl";
import { render, cleanup } from "@testing-library/react";
import { MuiThemeProvider, createTheme } from "@material-ui/core";

import { ConfirmationDialog } from "./index";

afterEach(cleanup);

const i18n = {
  "dialogs.confirmation": "Please confirm",
  "form.buttons.cancel": "Cancel",
  "form.buttons.ok": "Ok",
};

describe("<ConfirmationDialog />", () => {
  it("renders closed dialog", () => {
    const container = document.createElement("div");
    document.body.append(container);

    const props = {
      container,
      open: false,
      onConfirm: jest.fn(),
      onCancel: jest.fn(),
      children: "some text",
    };

    const { asFragment } = render(
      <MuiThemeProvider theme={createTheme()}>
        <IntlProvider locale="en" messages={i18n}>
          <ConfirmationDialog {...props} />
        </IntlProvider>
      </MuiThemeProvider>,
      { container },
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders open component", () => {
    const container = document.createElement("div");
    document.body.append(container);

    const props = {
      container,
      open: true,
      onConfirm: jest.fn(),
      onCancel: jest.fn(),
      children: "some text",
    };

    const { asFragment } = render(
      <MuiThemeProvider theme={createTheme()}>
        <IntlProvider locale="en" messages={i18n}>
          <ConfirmationDialog {...props} />
        </IntlProvider>
      </MuiThemeProvider>,
      { container },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
