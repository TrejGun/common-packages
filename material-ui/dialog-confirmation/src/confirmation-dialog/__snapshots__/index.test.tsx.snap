// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationDialog /> renders closed dialog 1`] = `<DocumentFragment />`;

exports[`<ConfirmationDialog /> renders open component 1`] = `
<DocumentFragment>
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-describedby="confirmation-dialog-content"
        aria-labelledby="confirmation-dialog-title"
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div
          class="MuiDialogTitle-root"
          id="confirmation-dialog-title"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6"
          >
            Please confirm
          </h2>
        </div>
        <div
          class="MuiDialogContent-root"
          id="confirmation-dialog-content"
        >
          <div
            class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            some text
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Cancel
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Ok
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;
